homeassistant/components/lacrosse_view/config_flow.py

```diff
@@ -68,11 +70,28 @@ async def async_step_user(
         except NoLocations:
             errors["base"] = "no_locations"
         except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception("Unexpected exception")
+            LOGGER.exception("Unexpected exception")
             errors["base"] = "unknown"
         else:
             self.data = user_input
             self.locations = info
+
+            # Check if we are reauthenticating
+            if self._reauth_data["id"] is not None:
```

Comment:
I'd instead save the existing entry as an instance attribute already in the reauth step.

https://github.com/home-assistant/core/blob/7ec54edc69c8f2563045b92a68370119d5b4aef2/homeassistant/components/openexchangerates/config_flow.py#L110-L115

https://github.com/home-assistant/core/blob/7ec54edc69c8f2563045b92a68370119d5b4aef2/homeassistant/components/openexchangerates/config_flow.py#L94-L99

Then we don't need an extra attribute `_reauth_data` for the existing config entry data.